### ISSUE TITLE
FR: MountFileSystem

### ISSUE THREAD

**Issue by tvolkert** [matanlurey/file#45]
Provides the ability to mount arbitrary file systems as "directories"

+cbracken, coupled with `ChrootFileSystem`, this would provide the functionality you were talking about a few weeks ago.

**Comment by tvolkert** [comment]
@cbracken

**Comment by tvolkert** [comment]
Also open to a better name... `Mountable`.... `Mounting`.... `Multi`....?  I kinda like `Multi`.

**Comment by cbracken** [comment]
Agreed I sort of like `Multi` too. We used `OverlayFileSystem` but that's pretty long. `Union` was a decent (and short) contender as well, though that doesn't really imply which one wins when you mount something. "Mount" has the advantage that, at least with POSIXy systems, the mounted one wins.

**Comment by matanlurey** [comment]
+1 for mount.
